### PR TITLE
Feature/cucumber tests

### PR DESCRIPTION
**Pull request by wandmagic**
Introduce yaml test runner

### CODE REVIEW

File: features/steps/fedramp_extensions_steps.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    // Add other pipeline steps as needed
+  }
+}else{
+  processedContentPath = contentPath;
+}
+
+  //Validate processed content
+  // Check expectations
+  const result =await Object.entries(testCase.expectations)
```
Comment (david-waltermire) [inline features/steps/fedramp_extensions_steps.ts:85]: Maybe rename `result` result to expectations or similar.

File: .github/workflows/content-artifacts.yml [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
+        - master
+  name: Process Content
+  env:
+    HOME_REPO: GSA/fedramp-automation
+  jobs:
+    validate-and-publish-content:
+      name: Content Validation Checking, Conversion and Validation
+      runs-on: ubuntu-20.04
+      steps:
+        - uses: actions/checkout@v2
```
Comment (david-waltermire) [inline .github/workflows/content-artifacts.yml:23]: This should be pinned to a specific commit hash.

File: .github/workflows/content-artifacts.yml [hunk 3]
```diff
@@ -1,55 +1,55 @@
...
+        - name: Setup SSH key
+          # Only do this on GSA/master
+          if: github.repository == env.HOME_REPO && github.ref == 'refs/heads/master'
+          run: |
+            eval "$(ssh-agent -s)"
+            ssh-add - <<< "${{ secrets.SSH_PRIVATE_KEY }}"
+        - name: Publish Artifacts
+          # Only do this on GSA/master
+          if: github.repository == env.HOME_REPO && github.ref == 'refs/heads/master'
+          uses: stefanzweifel/git-auto-commit-action@v4.4.1
```
Comment (david-waltermire) [inline .github/workflows/content-artifacts.yml:49]: This should be pinned to a specific commit hash.

File: .github/workflows/run-tests.yml [hunk 4]
```diff
@@ -35,36 +32,16 @@ jobs:
         uses: actions/setup-node@v1
```
Comment (david-waltermire) [inline .github/workflows/run-tests.yml:32]: This should be pinned to a specific commit hash.

File: .github/workflows/run-tests.yml [hunk 5]
```diff
@@ -13,12 +13,9 @@ on:
 jobs:
```
Comment (david-waltermire) [inline .github/workflows/run-tests.yml:13]: This is setup to run on `master`. Should we run this on all development, feature, and PR branches as well?

File: .github/workflows/run-tests.yml [hunk 6]
```diff
@@ -35,36 +32,16 @@ jobs:
...
         run: |
-          make test-validations test-web
+          make build-validations
 
       - name: Build all
         shell: bash
         if: runner.os == 'Windows' || runner.os == 'macOS'
         run: |
-          make build
+          make build
```
Comment (david-waltermire) [inline .github/workflows/run-tests.yml:47]: Does this include the validations? Is this in the right order with the previous steps?

File: @rerun.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+features/steps/fedramp_extensions.feature:4:13:30:31:32:33:34
```
Comment (david-waltermire) [inline @rerun.txt:1]: What is the purpose of this file?

File: src/content/rev5/baselines/xml/FedRAMP_rev5_HIGH-baseline_profile.xml [hunk 8]
```diff
@@ -3,50 +3,10 @@
...
-			<short-name>PMO</short-name>
-		</role>
-		<role id="fedramp-jab">
-			<title>The FedRAMP Joint Authorization Board (JAB)</title>
-			<short-name>JAB</short-name>
-		</role>
-		<party uuid="8cc0b8e5-9650-4d5f-9796-316f05fa9a2d" type="organization">
-			<name>Federal Risk and Authorization Management Program: Program Management Office</name>
-			<short-name>FedRAMP PMO</short-name>
-			<link href="https://fedramp.gov" rel="homepage"/>
```
Comment (david-waltermire) [inline src/content/rev5/baselines/xml/FedRAMP_rev5_HIGH-baseline_profile.xml:24]: Are these profile changes intended to be in this PR?

**Review by david-waltermire — CHANGES_REQUESTED** [review]
There are a few small refinements that I provided inline comments requesting.

There also appears to be some profile changes here, which looks like some of the work @Rene2mt is doing in a different branch. Could this be due to some merging/rebasing?

**Comment by wandmagic** [comment]
> There are a few small refinements that I provided inline comments requesting.
> 
> There also appears to be some profile changes here, which looks like some of the work @Rene2mt is doing in a different branch. Could this be due to some merging/rebasing?

yes i merged in the changes renee made to the profiles in this branch

File: @rerun.txt (hunk 7 above)
Comment (wandmagic) [reply @rerun.txt:1]: this has been removed, its an artifact that lets cucumber know which tests should be re-run that have failed

File: .github/workflows/run-tests.yml (hunk 6 above)
Comment (wandmagic) [reply .github/workflows/run-tests.yml]: i trimmed this, looks like it was running it twice

File: .github/workflows/run-tests.yml (hunk 4 above)
Comment (wandmagic) [reply .github/workflows/run-tests.yml]: done

File: .github/workflows/run-tests.yml (hunk 5 above)
Comment (wandmagic) [reply .github/workflows/run-tests.yml:16]: added develop and feature branches

File: src/content/rev5/baselines/xml/FedRAMP_rev5_HIGH-baseline_profile.xml (hunk 8 above)
Comment (david-waltermire) [reply src/content/rev5/baselines/xml/FedRAMP_rev5_HIGH-baseline_profile.xml:24]: Please rebase this branch to remove the merge commits. I can help you if you'd like some assistance.

**Comment by david-waltermire** [comment]
> > There are a few small refinements that I provided inline comments requesting.
> > There also appears to be some profile changes here, which looks like some of the work @Rene2mt is doing in a different branch. Could this be due to some merging/rebasing?
> 
> yes i merged in the changes renee made to the profiles in this branch

Please rebase this branch to remove the merge commits. I can help you if you'd like some assistance.

**Comment by wandmagic** [comment]
see #613 for an isolated only the yaml testing script pull request

**Pull request closed without merging**